src/d3a/d3a_core/sim_results/stats.py

```diff
@@ -119,14 +122,14 @@ def energy_bills(self, area, past_market_types):
                 buyer = area_name_from_area_or_iaa_name(trade.buyer)
                 seller = area_name_from_area_or_iaa_name(trade.seller)
                 if buyer in result:
-                    self._store_bought_trade(result[buyer], trade.offer)
+                    self._store_bought_trade(result[buyer], trade)
                 if seller in result:
-                    self._store_sold_trade(result[seller], trade.offer)
+                    self._store_sold_trade(result[seller], trade)
         for child in area.children:
-            child_result = self.energy_bills(child, past_market_types)
+            child_result = self._energy_bills(child, past_market_types)
             if child_result is not None:
                 result[child.name]['children'] = child_result
-                result[child.name]['market_fee'] = self.market_fees[child.name]
+                # result[child.name]['market_fee'] = self.market_fees[child.name]
```

Comment:
Could you also add either to the unit tests or to the integration tests an assert that fees are populated on the energy_bills? I am not sure, but it would be nice to have an extra check that the grid fees end up in the flattened results of the energy bills.

Reply:
Correct me if Im wrong but isn't this [test](https://github.com/gridsingularity/d3a/blob/bug/D3ASIM-1862/tests/test_stats.py#L214) is already taking care of it?

Reply:
I am not sure if it is really needed, but we do not see the market fee for the grid market in the test you have linked to.